### ISSUE TITLE
Score becomes 0 if no MS2 peaklist information provide.

### ISSUE THREAD

**Issue by xiaodfeng** [c-ruttkies/MetFragR#25]
Hi,
We have generated a dataset with only MS1, when I try to perform the database searching without MS2, all the probabilistic scores become 0. How can we deal with this issue?

**Comment by schymane** [comment]
The MetFrag scoring function needs fragments (MS2) to come up with a score above zero. If you do not have any fragmentation data, then you can only use additional metadata (literature, patent counts, etc) to score your candidates. If you are using PubChem, then you can retrieve this additional data and add it as a score on the "Candidate Filter and Scoring Settings" tab. There you will also find other available options to score your candidates. But please do bear in mind that most additional scores are not based on experimental data then, so e.g. the ranking based on patent and literature counts may not always make sense in terms of your experimental context.

**Comment by schymane** [comment]
https://msbi.ipb-halle.de/MetFrag/
and
https://github.com/schymane/ReSOLUTION
have more information on scoring functions. 
ReSOLUTION still needs a peak list to run, but you can set a dummy MS/MS file of mz 999 and int 999 so that the fragmenter score is always zero, but other metadata scores can be used.

**Comment by xiaodfeng** [comment]
@schymane Thank you for the explaination. I have also tried using additional metadata (literature, patent counts, etc). But the score is still 0, as the MetFrag score mostly based on the fragments score. Under this case, the package ReSOLUTION is a better choice. I have installed ReSOLUTION. But the only manual I have found is https://github.com/schymane/ReSOLUTION/blob/master/man/runMetFrag.Rd.
Is there any detailed description of this package?

**Comment by schymane** [comment]
The individual functions are also documented (this can be browsed via the help functions); there is no other documentation yet as I usually train up people in house to use this (as the functionality is quite complicated). We are working on this, I am just building my group and training up people, we will release more documentation etc as we are able/ready to do so.

**Comment by xiaodfeng** [comment]
@schymane I have tried ReSOLUTION for identification by setting a dummy MS/MS file of mz 999 and int 999. However, the following scores are all zero.
`Score, OfflineMetFusionScore, FragmenterScore, and OfflineIndividualMoNAScore.`
Under this case, the solution I have thought is to add a new `MS1 mass difference` column for scoring.
Of course, this is not a perfect solution.

**Comment by schymane** [comment]
Yes, all these scores will be zero, because they require MS/MS peaks to get a non-zero score. You will need to add other scores (e.g. PubChem patents, literature references, CompTox metadata, retention time scores) to get any non-zero score. The MS1 mass difference will be the same for all candidates with the same molecular formula, so it would not be very helpful beyond maybe indicating which formulas are closer to the measured mass than others.
